web/src/components/metrics/metrics-content.tsx

```diff
@@ -0,0 +1,330 @@
+
+          return data?.namespace
+            ? m.SrcK8S_Namespace === data.name
+              ? `${t('To')} ${dstNamespace}`
+              : `${t('From')} ${srcNamespace}`
+            : `${srcNamespace} -> ${dstNamespace}`;
+        case MetricScopeOptions.OWNER:
+          const srcOwner = truncateParts(getMetricName(m, scope, true, t));
+          const dstOwner = truncateParts(getMetricName(m, scope, false, t));
+
+          return data?.namespace
+            ? m.SrcK8S_Namespace === data.namespace
+              ? `${t('To')} ${dstOwner}`
+              : `${t('From')} ${srcOwner}`
+            : `${srcOwner} -> ${dstOwner}`;
+        case MetricScopeOptions.RESOURCE:
+        default:
+          const src = truncateParts(getMetricName(m, scope, true, t));
+          const dst = truncateParts(getMetricName(m, scope, false, t));
+
+          return data?.addr
+            ? m.SrcAddr === data.addr
+              ? `${t('To')} ${dst}`
+              : `${t('From')} ${src}`
+            : `${src} -> ${dst}`;
+      }
+    }
+
+    const total = metrics.reduce((prev, cur) => prev + cur.total, 0);
+
+    const legendData = metrics.map(m => ({
+      childName: `${showBar ? 'bar-' : 'area-'}${metrics.indexOf(m)}`,
+      name: getName(m.metric)
+    }));
+
+    const CursorVoronoiContainer = createContainer('voronoi', 'cursor');
+
+    const containerComponent = (
+      <CursorVoronoiContainer
+        cursorDimension="x"
+        labels={({
+          datum
+        }: {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          datum: any;
+        }) => `${datum.y !== null ? datum.y : 'no data'}`}
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        labelComponent={<ChartLegendTooltip legendData={legendData} title={(datum: any) => datum.x} />}
+        mouseFollowTooltips
+        voronoiDimension="x"
+        voronoiPadding={50}
+      />
+    );
+
+    const legentComponent = (
+      <ChartLegend
+        labelComponent={<ChartLabel className={smallerTexts ? 'small-chart-label' : ''} />}
+        data={legendData}
+      />
+    );
+
+    return (
+      <div
+        id={`chart-${id}`}
+        style={{
+          width: sizePx ? `${doubleWidth ? 2 * sizePx : sizePx}px` : '100%',
+          height: sizePx ? `${sizePx}px` : '100%',
+          alignSelf: 'center'
+        }}
+      >
+        {showDonut ? (
+          <ChartDonut
+            themeColor={ChartThemeColor.multiUnordered}
+            constrainToVisibleArea
+            legendData={legendData}
+            legendOrientation="vertical"
+            legendPosition="right"
+            legendAllowWrap={true}
+            legendComponent={legentComponent}
+            labels={({ datum }) => datum.x}
+            width={doubleWidth ? 1000 : 500}
+            height={350}
+            data={metrics
+              .sort((a, b) => a.total - b.total)
```

Comment:
@jpinsonneau not 100% sure but I think this sort messes up with the legend, as I think it relies on values orders to match with metric info

Reply:
I'll double check that :+1: